iib/workers/tasks/build.py

```diff
@@ -764,26 +646,22 @@ def _prepare_request_for_build(
     # Use the distribution_scope of the from_index as the resolved distribution scope for `Add`,
     # and 'Rm' requests, but use the distribution_scope of the target_index as the resolved
     # distribution scope for `merge-index-image` requests.
-    resolved_distribution_scope = from_index_info['resolved_distribution_scope']
-    if source_from_index:
-        resolved_distribution_scope = target_index_info['resolved_distribution_scope']
+    resolved_distribution_scope = index_info['from_index']['resolved_distribution_scope']
+    if build_request_config.source_from_index:
+        resolved_distribution_scope = index_info['target_index']['resolved_distribution_scope']
 
     distribution_scope = _validate_distribution_scope(
-        resolved_distribution_scope, distribution_scope
+       resolved_distribution_scope, build_request_config.distribution_scope 
     )
 
-    if not binary_image:
-        binary_image_ocp_version = from_index_info['ocp_version']
-        if source_from_index:
-            binary_image_ocp_version = target_index_info['ocp_version']
-
-        binary_image = get_binary_image_from_config(
-            binary_image_ocp_version, distribution_scope, binary_image_config
-        )
+    binary_image = build_request_config.get_binary_image(
+        index_info['from_index'], distribution_scope
+    )
 
     binary_image_resolved = get_resolved_image(binary_image)
     binary_image_arches = _get_image_arches(binary_image_resolved)
 
+    print(arches, binary_image_arches)
```

Comment:
Let's remove this :slightly_smiling_face:

Reply:
ah, debug print I forgot to remove